src/ResourceManager/Version2016_09_01/AzureRMCmdlet.cs

```diff
@@ -86,7 +86,56 @@ public IAzureContextContainer DefaultProfile
             set
             {
                 _profile = value;
+                var accountMap = CreateAccountMap(AzureRmProfileProvider.Instance.Profile.Accounts);
```

Comment:
Please add comment to here. We need to explain why we need to load secret from context.